### PR TITLE
upload TheXTech engine port

### PR DESCRIPTION
**Pull request by ddrsoul**
# New Port for TheXTech engine (opensource SMBX port)

## Game Information
- **Title**: TheXTech
- **URL**: https://github.com/TheXTech/TheXTech

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [X] ArkOS
- [X] RetroOZ
- [X] ROCKNIX
- [X] MUOS
- [X] KNULLI

### Resolution Tests
Test all major resolutions:
- [X] 480x320
- [X] 640x480
- [X] 854x480
- [ ] 720x720 (RGB30)
- [X] Higher resolutions 1920x1080

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

File: ports/thextech/README.MD [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+make
+```
+
+
+## HOW TO RUN
+
+Package contains only open source engine, for gaming you need assets (sprites, music and etc.), you can find them at the original github project https://github.com/TheXTech/TheXTech
+add assets into
+```
+./thextech/assets/assets_name_folder
```
Comment (kloptops) [inline ports/thextech/README.MD:34]: Create this directory with an empty text file named `./thextech/assets/assets_name_folder/.gitkeep` so that its easier for people to add assets inside.

File: ports/thextech/README.MD [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+## ROCKNIX Powkiddy X55 gamepad fix
+
+There are reports that TheXTech has problems detecting gamepad for this particular device. Solution is to edit TheXTech.sh line#34
+```
+$GPTOKEYB "thextech" -c "$GAMEDIR/thextech.gptk" &
+```
+replace with 
+```
+$GPTOKEYB "thextech" xbox360 &
+```
```
Comment (kloptops) [inline ports/thextech/README.MD:79]: Can't this be done via bash?

**Review by kloptops — CHANGES_REQUESTED** [review]

File: ports/thextech/README.MD (hunk 1 above)
Comment (ddrsoul) [reply ports/thextech/README.MD:34]: the thing is that assets name folder can be different , so name can be different

Comment (kloptops) [reply ports/thextech/README.MD:34]: Then maybe reword so it is more clear:

Package contains only open source engine, for gaming you need assets (sprites, music and etc.), you can find them at the original github project https://github.com/TheXTech/TheXTech add assets into `thextech/assets/<some_folder>`, for example:
- `thextech/assets/<something>/`

On first run it will do blah and create:
- `thextech/assets/<something else>/`

File: ports/thextech/README.MD (hunk 2 above)
Comment (ddrsoul) [reply ports/thextech/README.MD:79]: I have a feeling that this might be fixed by adding
```
export SDL_GAMECONTROLLERCONFIG="$sdl_controllerconfig"
$ESUDO chmod 666 /dev/uinput
```
but I have nobody to test. I suggest to not add complicated stuff into .sh and leave as it is now to get more feedback. I will update this package anyway when TheXTech 1.3.7 will get into beta stage (now it is pre-beta :D )

File: ports/thextech/README.MD (hunk 1 above)
Comment (ddrsoul) [reply ports/thextech/README.MD]: I made changes in readme, I hope it is more understandable now

Comment (ddrsoul) [reply ports/thextech/README.MD]: Another change in Readme, now we agreed that it is ok in Discord!

File: ports/thextech/README.MD (hunk 2 above)
Comment (ddrsoul) [reply ports/thextech/README.MD:82]: My idea was to add
```
if [${DEVICE_NAME} == "X55" && ${CFW_NAME} == "ROCKNIX"]; then
  $GPTOKEYB "thextech" xbox360 &
else
  $GPTOKEYB "thextech" -c "$GAMEDIR/thextech.gptk" &
fi
```
into TheXTech.sh but I can't test it and nobody cares about X55. I think for one that is missing there is no spoiling a wedding

File: ports/thextech/screenshot.png
Comment (kloptops) [inline ports/thextech/screenshot.png:1]: Screenshot should be 640x480. :)

**Review by kloptops — CHANGES_REQUESTED** [review]
Just need to fix the screenshot i think.

File: ports/thextech/screenshot.png
Comment (ddrsoul) [reply ports/thextech/screenshot.png:1]: thanks! new screenshot uploaded

**Review by Cebion — APPROVED** [review]

**Review by kloptops — APPROVED** [review]
